third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1254,6 +1261,8 @@ pub enum SignatureToken {
     Signer,
     /// Vector
     Vector(Box<SignatureToken>),
+    /// Function, with n argument types and m result types, and an associated ability set.
+    Function(Vec<SignatureToken>, Vec<SignatureToken>, AbilitySet),
```

Comment:
Using a struct-style enum variant here causes those mysterious Clippy problems.  I think it's related to one of the `derive` macros on `SignatureToken` declaration (probably `dearbitrary::Dearbitrary`) lacking handling of struct-style enum variant.  In the interest of moving faster, I'm going back to the tuple-style enum variant instead, rather than trying to fix that 3d-party code. :-)